src/System.CommandLine/Collections/SymbolSet.cs

```diff
@@ -91,5 +111,51 @@ item switch
                 IIdentifierSymbol named => named.Aliases,
                 _ => new[] { item.Name }
             };
+
+        internal IReadOnlyList<Argument> Arguments
+        {
+            get
+            {
+                return _arguments ??= BuildArgumentsList();
+
+                List<Argument> BuildArgumentsList()
+                {
+                    var arguments = new List<Argument>(Count);
+
+                    for (var i = 0; i < Count; i++)
+                    {
+                        if (this[i] is Argument argument)
+                        {
+                            arguments.Add(argument);
+                        }
+                    }
+
+                    return arguments;
+                }
+            }
+        }
+        
+        internal IReadOnlyList<Option> Options
+        {
+            get
+            {
+                return _options ??= BuildOptionsList();
+
+                List<Option> BuildOptionsList()
+                {
+                    var options = new List<Option>(Count);
+
+                    for (var i = 0; i < Count; i++)
+                    {
+                        if (this[i] is Option Option)
+                        {
+                            options.Add(Option);
+                        }
+                    }
+
+                    return options;
+                }
+            }
+        }
```

Comment:
If these are read-only lists, consider storing them as `ReadOnlyCollection<T>` to avoid messing with the collection.

The empty collection could be a static reused.

Reply:
There's no clear way to initialize the `ReadOnlyCollection<T>` without introducing an intermediate enumerator, so it won't save us anything.